server/etcdserver/api/v3rpc/validationfuzz_test.go

```diff
@@ -0,0 +1,219 @@
+package v3rpc
+
+import (
+	"context"
+	"testing"
+
+	pb "go.etcd.io/etcd/api/v3/etcdserverpb"
+	txn "go.etcd.io/etcd/server/v3/etcdserver/txn"
+	"go.etcd.io/etcd/server/v3/lease"
+	betesting "go.etcd.io/etcd/server/v3/storage/backend/testing"
+	"go.etcd.io/etcd/server/v3/storage/mvcc"
+	"go.uber.org/zap/zaptest"
+)
+
+func FuzzRangeRequest(f *testing.F) {
+	testcases := []pb.RangeRequest{
+		{
+			Key:        []byte{2},
+			RangeEnd:   []byte{2},
+			Limit:      3,
+			Revision:   3,
+			SortOrder:  2,
+			SortTarget: 2,
+		},
+	}
+
+	for _, tc := range testcases {
+		soValue := pb.RangeRequest_SortOrder_value[tc.SortOrder.String()]
+		soTarget := pb.RangeRequest_SortTarget_value[tc.SortTarget.String()]
+		f.Add(tc.Key, tc.RangeEnd, tc.Limit, tc.Revision, soValue, soTarget) // Use f.Add to provide a seed corpus
+	}
+	f.Fuzz(func(t *testing.T,
+		key []byte,
+		rangeEnd []byte,
+		limit int64,
+		revision int64,
+		sortOrder int32,
+		sortTarget int32,
+	) {
+		b, _ := betesting.NewDefaultTmpBackend(t)
+		defer betesting.Close(t, b)
+		s := mvcc.NewStore(zaptest.NewLogger(t), b, &lease.FakeLessor{}, mvcc.StoreConfig{})
+		defer s.Close()
+
+		// setup cancelled context
+		ctx, cancel := context.WithCancel(context.TODO())
+		cancel()
+		// put some data to prevent early termination in rangeKeys
+		// we are expecting failure on cancelled context check
+		s.Put(key, []byte("bar"), lease.NoLease)
+
+		request := &pb.TxnRequest{
+			Success: []*pb.RequestOp{
+				{
+					Request: &pb.RequestOp_RequestRange{
+						RequestRange: &pb.RangeRequest{
+							Key:        key,
+							RangeEnd:   rangeEnd,
+							Limit:      limit,
+							SortOrder:  pb.RangeRequest_SortOrder(sortOrder),
+							SortTarget: pb.RangeRequest_SortTarget(sortTarget),
+						},
+					},
+				},
+			},
+		}
+		errCheck := checkRangeRequest(&pb.RangeRequest{
+			Key:        key,
+			RangeEnd:   rangeEnd,
+			Limit:      limit,
+			SortOrder:  pb.RangeRequest_SortOrder(sortOrder),
+			SortTarget: pb.RangeRequest_SortTarget(sortTarget),
+		})
+
+		if errCheck != nil {
+			t.Skip("Validation not passing. Skipping the apply.")
+		}
+
+		_, _, err := txn.Txn(ctx, zaptest.NewLogger(t), request, false, s, &lease.FakeLessor{})
+		if err != nil {
+			t.Logf("Check: %s | Apply: %s", errCheck, err)
```

Comment:
Shouldn't we fail the test?

Reply:
The goal of the test is to verify that the `Tnx` function won't panic in case the validation check passes.  it is okay in this context if the `Tnx` handles the request with an error. In case of panic, the test should fail.

Reply:
OK.